app/models/user.rb

```diff
@@ -10,10 +10,23 @@ class User < ActiveRecord::Base
 
   def self.from_omniauth(auth)
     find_or_create_by(provider: auth.provider, uid: auth.uid) do |user|
-      user.provider = auth.provider
-      user.uid = auth.uid
-      user.email = auth.info.email
-      user.password = Devise.friendly_token[0,20]
+      user.provider   = auth.provider
+      user.uid        = auth.uid
+      user.email      = auth.info.email
+      user.password   = Devise.friendly_token[0,20]
     end
   end
+
+  def self.current_id=(id)
```

Comment:
Assing current_user into Thread, so that we can access `User.current` outside `Controller`. Check out what I've done inside `Review` on `before_save`